File: src/ConsoleHelper.java
public class ConsoleHelper {
    private String lastLine = "";

    public void print(String line) {
        //clear the last line if longer
        if (lastLine.length() > line.length()) {
            String temp = "";
            for (int i = 0; i < lastLine.length(); i++) {
                temp += " ";
            }
            if (temp.length() > 1)
                System.out.print("\r" + temp);
        }
        System.out.print("\r" + line);
        lastLine = line;
    }

    private byte anim;

    public void animate() {
        switch (anim) {
            case 1:
                print("[ \\ ] ");
                break;
            case 2:
                print("[ | ] ");
                break;
            case 3:
                print("[ / ] ");
                break;
            default:
                anim = 0;
                print("[ ─ ] ");
        }
        anim++;
    }
}

File: src/ConsoleHelper.kt
class ConsoleHelper {
    private var lastLine = ""
    private fun print(line: String) {
        //clear the last line if longer
        if (lastLine.length > line.length) {
            var temp = ""
            for (i in lastLine.indices) {
                temp += " "
            }
            if (temp.length > 1) kotlin.io.print("\r" + temp)
        }
        kotlin.io.print("\r" + line)
        lastLine = line
    }

    private var anim: Int = 0
    fun animate() {
        when (anim) {
            1 -> print("[ \\ ] ")
            2 -> print("[ | ] ")
            3 -> print("[ / ] ")
            else -> {
                anim = 0
                print("[ ─ ] ")
            }
        }
        anim++
    }
}